### ISSUE TITLE
Add top metro regions to state lookups?

### ISSUE THREAD

**Issue by knowtheory** [censusreporter/censusreporter#113]
I was trying to answer questions for my inlaws regarding largest cities and such in the various states on our trip from Ohio down to Alabama, and realized that Wikipedia is *terrible* for answering these sorts of questions.  For example, Ohio's wikipedia page has [a Major Cities section](http://en.wikipedia.org/wiki/Ohio#Major_cities), which lists top 10 cities by municipality, placing Columbus as the most populous city in Ohio, despite the fact that if you compare the [Columbus](http://censusreporter.org/profiles/31000US18140-columbus-oh-metro-area/), [Cleveland](http://censusreporter.org/profiles/31000US17460-cleveland-elyria-oh-metro-area/) and [Cincinnati](http://censusreporter.org/profiles/31000US17140-cincinnati-oh-ky-in-metro-area/) metro regions, Columbus is the smallest of the three (granted, by a narrow margin).

I was happy to be able to refer to CensusReporter to quickly do that comparison for Ohio, but doing the same for states one is less familiar with gets harder without having to cross reference what metro regions exist in a particular state first.

Anyway.  Wondering whether adding such a display would be interesting/reasonable; whether adding a display for just one type of region (state) is architecturally feasible (writing code for exceptional kinds of things is always a bummer); and whether this is just a crazy thing to ask about :)

**Comment by knowtheory** [comment]
Oh, there was an additional architectural caveat, which is that in cases like Cincinnati, the metro region spans three states.  So when one says "what are the top 5 metro regions in Ohio", the answer to one's Q would depend on whether one is focused on stuff exclusively in state, or about a whole metro region that extends into the state in question.

**Comment by JoeGermuska** [comment]
@knowtheory not crazy at all! 

So, first, you can select "metro areas in [state]" as a geography for a table, as in:
![select metro areas](https://cloud.githubusercontent.com/assets/46313/7554723/f8ad617c-f6f8-11e4-89bb-831955e77b1c.png)

![select a container for metro areas](https://cloud.githubusercontent.com/assets/46313/7554724/1179ab5c-f6f9-11e4-9170-52cb5dab6ee4.png)

which leads to [Total population of metro areas in Tennessee](http://censusreporter.org/data/table/?table=B01003&geo_ids=310|04000US47&primary_geo_id=04000US39)

Similarly, you can divide any primary geography with this part of the "data" view:

![image](https://cloud.githubusercontent.com/assets/46313/7554748/08e9a3e2-f6fa-11e4-8565-334fa5ffb519.png)

In this implementation, it selects metros with any intersection with the given state. Better to have too many than too few, I think.

Of course, the relentless horizontality of that view is at odds with the use case here, so a "list view" could be good -- we talked about it some, but didn't arrive at a satisfying design that took priority over other features. You could imagine it as a "column" view, although I would definitely consider the idea of less generic implementations that serve a specific use case really well. (That is, something a bit more curated like the profile pages are.)

If you're interested in trying to contribute, that's very welcome and of course I'll help with setup, etc. But I think we should hash out design before starting anything. 

cc @ryanpitts @SaraSchnadt

**Comment by knowtheory** [comment]
Ohhhhhh.  That is so super cool!

I don't think i would have been able to find that on my own.  I knew that Census Reporter had table displays, but even knowing that, i didn't see a way to find the table displays just from a scan of the state's page.

![screen shot 2015-05-10 at 11 11 53 am](https://cloud.githubusercontent.com/assets/192/7554855/aa02a8f8-f706-11e4-8d4b-8c921ec027c6.png)

I didn't immediately think to flip open the "view data" for the demographics which has the "view table" link.  Would there be a way to include a "view tables" link under the Demographics label on the left or something?

The intersection / reaggregation is SO COOL.  The map plot is a pretty good proxy for what i was interested in (which you're right is the useful display here).  Btw, the fact that the headers/columns aren't a scrollable area on the table display is a little weird/confusing when there's only one row of data (the scrollable area / target is pretty small, and the headers end up being much larger than the data).

And yep, i'd be happy to try and pitch something in and talk about designy things :)

**Comment by JoeGermuska** [comment]
> Would there be a way to include a "view tables" link under the Demographics label on the left or something?

Are you thinking something like this:

![prospective "tables in this section" image](https://cloud.githubusercontent.com/assets/46313/7554974/83f7b6b2-f701-11e4-92dc-b1de98b1d5bb.png)

It might be more crowded in other sections, and in fact, there are multiple rows under each header, so having the "Tables used" header could get repetitive. (although also the same table is sometimes used in more than one row.)

Obviously this page is heavily crafted, so we can work around those things. I'm not sure I see the compelling value, but open to iterations and discussion.

> the fact that the headers/columns aren't a scrollable area on the table display is a little weird/confusing when there's only one row of data (the scrollable area / target is pretty small, and the headers end up being much larger than the data).

This is part of why I'm open to designing a list view, and to the idea that it might be more designed than the super-generalized-work-with-any-combination-of-geographies-and-any-table `/data/` view. We might see if there's anything to learn from the somewhat obscure "ranking tables" which the Bureau releases. They aren't in bulk download, but here's one example, [States ranked by pctg of population that is white alone](http://factfinder.census.gov/bkmk/table/1.0/en/ACS/13_1YR/R0201.US01PRF). It doesn't look like you can get these for metro areas, although I don't know why.

**Comment by knowtheory** [comment]
> Are you thinking something like this

Yep quite like that.  (although would have to give some thought to how that would fold up when viewed on a mobile browser.

Part of my issue i guess is even knowing what tables are available for me to pivot/reaggregate upon.  The [table codes page](http://censusreporter.org/topics/table-codes/) kinda fits the bill, but i totally lost it in [the list of topics on the front page](http://censusreporter.org/#uv-1).  The list is a little... grab-bag-y so i guess i skimmed over it and missed the [Getting Started](http://censusreporter.org/topics/getting-started/) and [table codes](http://censusreporter.org/topics/table-codes/) pages.

Ultimately, I think it comes down to the fact that i don't have any expectations for what goes into this box, or what it'll do if i do enter stuff into it:

![screen shot 2015-05-12 at 11 05 08 am](https://cloud.githubusercontent.com/assets/192/7592032/d295f176-f896-11e4-85c0-c41c13ea3616.png)

Having some expectation that i can reaggregate once i land on a table display is super powerful/cool.  I almost want something like [Zapier's mashup interface](https://www.youtube.com/watch?v=04tffSuqFpk#t=16) which would give me a very explicit notion that i should pick some geographies and jam it together with some data.

Also looks like i've gotten CensusReporter set up on my machine fine. :)

**Comment by JoeGermuska** [comment]
I'm not terribly persuaded by the margin listing of tables referenced. Seems more like clutter than benefit.

Totally agree that knowing which tables are available is tricky and needs improvement. There's [a ticket](https://github.com/censusreporter/censusreporter/issues/111) for doing better-than-alphabetic ordering on the 'topical help' pages. I recently implemented a faceted browser for a project for a friend partly as a way to get a handle on what would be needed to make a faceted table browser for CR. I haven't had much time to move that forward, because even extracting the facets is a little trickier than I'd hoped. (If you feel like hacking on that, let me know!)

We've mostly been operating on the idea that a few YouTube videos could be the most effective way to train people how to use Census Reporter. I'd prefer not to require videos, but we're walking a fine line between a tool for experts and a tool for everyone. If you're referring to stuff like Zapier's "handwritten" labeled fields, I like the general idea but haven't specifically thought about how to do similar for Census Reporter.

The "Find data for this place" is obscure, yes. It's also something we didn't figure many people would use, although we haven't done anything to make it more usable for those who want-it-but-don't-know-that's-what-it-does.
